File: src/main/java/trivia/QuestionDeck.java
package trivia;

import java.util.LinkedList;

public class QuestionDeck {

    private static final int QUESTIONS_PER_CATEGORY = 50;

    private final LinkedList<String> popQuestions = new LinkedList<>();
    private final LinkedList<String> scienceQuestions = new LinkedList<>();
    private final LinkedList<String> sportsQuestions = new LinkedList<>();
    private final LinkedList<String> rockQuestions = new LinkedList<>();
    private final LinkedList<String> geographyQuestions = new LinkedList<>();

    public QuestionDeck() {
        for (int i = 0; i < QUESTIONS_PER_CATEGORY; i++) {
            popQuestions.addLast("Pop Question " + i);
            scienceQuestions.addLast("Science Question " + i);
            sportsQuestions.addLast("Sports Question " + i);
            rockQuestions.addLast("Rock Question " + i);
            geographyQuestions.addLast("Geography Question " + i);
        }
    }

    public String nextQuestion(String category) {
        switch (category) {
            case "Pop":       return popQuestions.removeFirst();
            case "Science":   return scienceQuestions.removeFirst();
            case "Sports":    return sportsQuestions.removeFirst();
            case "Geography": return geographyQuestions.removeFirst();
            default:          return rockQuestions.removeFirst();
        }
    }
}

File: python/question_deck.py
from collections import deque


class QuestionDeck:
    """Owns and serves questions for every category, including Geography."""

    QUESTIONS_PER_CATEGORY = 50
    CATEGORIES = ("Pop", "Science", "Sports", "Geography", "Rock")

    def __init__(self) -> None:
        self._deck: dict[str, deque[str]] = {
            category: deque(
                f"{category} Question {i}"
                for i in range(self.QUESTIONS_PER_CATEGORY)
            )
            for category in self.CATEGORIES
        }

    def next_question(self, category: str) -> str:
        """Return and remove the next question for the given category."""
        return self._deck[category].popleft()
